### PR TITLE
gives functionality to login/signup forms; saves custom user data to database on sign up; reroutes to chat.html while logged in as currentuser

### PR DESCRIPTION
**Pull request by nlumiere**

### CODE REVIEW

**Review by arringtonh — APPROVED** [review]
lgtm

File: proj/src/main/webapp/script.js [hunk 1]
```diff
@@ -29,6 +30,7 @@ if(btnLogin){
 // Add sign up event
 if(btnSignUp){
     btnSignUp.addEventListener("click", e => {
+        const dbRef = firebase.database().ref("/users");
```
Comment (aokeke) [inline proj/src/main/webapp/script.js]: I don't think this is being used.

File: proj/src/main/webapp/script.js [hunk 2]
```diff
@@ -37,12 +39,23 @@ if(btnSignUp){
...
 
         const promise = auth.createUserWithEmailAndPassword(emailVal, passVal).then(function(){
-            auth.currentUser.updateProfile({displayName: fname.value + " " + lname.value}).then(function(){
+            const user = auth.currentUser;
+            user.updateProfile({
+                displayName: fname.value + " " + lname.value
+                }).then(function(){
                 console.log("display name updated successfully");
-                window.location.replace("chat.html");
+                if(fname && lname && tagStr){
```
Comment (aokeke) [inline proj/src/main/webapp/script.js]: Does this mean we don't want to store tagStr if they didn't enter a last name for example? Or that we don't want to store their name if they didn't enter any tags? Maybe you can just set the fields that are present, or if all is required, i'd do some kind of validation before creating the user.

We previously set their display name using their fname and lname so I would think up there we might want to also check if fname and lname is populated.

**Review by aokeke — CHANGES_REQUESTED** [review]

File: proj/src/main/webapp/script.js [hunk 3]
```diff
@@ -37,12 +38,21 @@ if(btnSignUp){
...
-                window.location.replace("chat.html");
+                firebase.database().ref("users/" + auth.currentUser.uid).set({
+                    firstName : fname.value,
+                    lastName : lname.value,
+                    tags : tagStr.value.split(',')
+                }).then(function(){
+                    window.location.replace("chat.html");
+                });
             }).catch(function(){
                 console.log("error updating display name");
```
Comment (aokeke) [inline proj/src/main/webapp/script.js:54]: Normally I'd suggest removing these console debug logs, but in this case, maybe you want to log something more descriptive while you debug. Like if the catch function can take a parameter with the error, you can log the error given.

In the finished product, you might want some sort of pop up with a user friendly error that says that an error occurred and suggesting they try again or something.

**Review by aokeke — APPROVED** [review]
LGTM 1 comment.

File: proj/src/main/webapp/script.js (hunk 3 above)
Comment (nlumiere) [reply proj/src/main/webapp/script.js:54]: Yup, that's what I was planning. Sounds good!

**Pull request merged**